newrelic/hooks/logger_structlog.py

```diff
@@ -50,16 +47,28 @@ def wrap__process_event(wrapped, instance, args, kwargs):
         settings = global_settings()
 
     # Return early if application logging not enabled
-    if settings and settings.application_logging and settings.application_logging.enabled:
-        if settings.application_logging.local_decorating and settings.application_logging.local_decorating.enabled:
-            event = add_nr_linking_metadata(event)
+    if settings and settings.application_logging.enabled:
```

Comment:
We probably can remove this `if settings` check since settings should be supplied somehow in the if check above. For an extra layer of safety in case `global_settings` is empty for some reason, we could also just do a `if settings` check before all this and return out of the wrapper if we don't get anything back.

Reply:
We want to guard against transaction settings being set to None, which has happened in the past.